wifiphisher/extensions/lure10.py

```diff
@@ -6,101 +6,108 @@
 Location Service
 """
 
-import struct
 import wifiphisher.common.constants as constants
 import scapy.layers.dot11 as dot11
 
 
 class Lure10(object):
     """
-    Sends a number of beacons to fool Windows Location Service.
+    Sends a number of beacons to fool Windows Location Service
     """
 
     def __init__(self, shared_data):
         """
-        Setup the class with all the given arguments.
+        Setup the class with all the given arguments
 
-        :param self: A Lure10 object.
+        :param self: A Lure10 object
         :param data: Shared data from main engine
         :type self: Deauthentication
-        :type data: dictionary
+        :type data: dict
         :return: None
         :rtype: None
         """
 
-        self.first = True
-        self.first_output = True
+        self.first_run = True
         self.data = shared_data
-        self.beacons_num = 0
 
     def get_packet(self, pkt):
         """
-        We start broadcasting the beacons on the first received packet.
+        We start broadcasting the beacons on the first received packet
 
-        :param self: A Lure10 object.
-        :param packet: A scapy.layers.RadioTap object.
+        :param self: A Lure10 object
+        :param packet: A scapy.layers.RadioTap object
         :type self: Lure10
         :type packet: scapy.layers.RadioTap
-        :return: list with the crafted beacon frames
-        :rtype: list
+        :return: A tuple containing ["*"] followed by a list of
+            the crafted beacon frames
+        :rtype: tuple(list, list)
+        .. warning: pkt is not used here but should not be removed since
+            this prototype is requirement
         """
 
-        beacons = []
-
-        if self.first:
-            if self.data.args.lure10_exploit:
-                area_file = constants.LOCS_DIR + self.data.args.lure10_exploit
-                with open(area_file) as a_file:
-                    wlans = [x.strip() for x in a_file.readlines()]
```

Comment:
Instead of using multiple loops to first read each line and then do the operations, they are combined in a single loop for better performance.